cmd/README.md

```diff
@@ -0,0 +1,738 @@
+	// 2) If BlockTimestamp is 0, this will be called while setting up the genesis block
+	// 3) If BlockTimestamp is 1000, this will be called while processing the first block
+	// whose timestamp is >= 1000
+	//
+	// Set the initial value under [common.BytesToHash([]byte("storageKey")] to "Hello World!"
+	res := common.LeftPadBytes([]byte("Hello World!"), common.HashLength)
+	state.SetState(HelloWorldAddress, common.BytesToHash([]byte("storageKey")), common.BytesToHash(res))
+}
+```
+We also see a `Verify()` function. We can leave this as is for now. 
+
+``` go
+// Verify tries to verify HelloWorldConfig and returns an error accordingly.
+func (c *HelloWorldConfig) Verify() error {
+
+	// CUSTOM CODE STARTS HERE
+	// Add your own custom verify code for HelloWorldConfig here
+	// and return an error accordingly
+	return nil
+}
+```
+
+Next place to modify is in our `sayHello()` function.
+
+``` go
+func sayHello(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SayHelloGasCost); err != nil {
+		return nil, 0, err
+	}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+	// no input provided for this function
+
+	// CUSTOM CODE STARTS HERE
+	// Get the current state
+	currentState := accessibleState.GetStateDB()
+	// Get the value set at recipient
+	value := currentState.GetState(HelloWorldAddress, common.BytesToHash([]byte("storageKey")))
+	// Do some processing and pack the output
+	packedOutput, err := PackSayHelloOutput(string(common.TrimLeftZeroes(value.Bytes())))
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+```
+
+Finally we can modify our `setGreeting()` function. 
+
+``` go
+func setGreeting(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SetGreetingGasCost); err != nil {
+		return nil, 0, err
+	}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+	// attempts to unpack [input] into the arguments to the SetGreetingInput.
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+    inputStr, err := UnpackSetGreetingInput(input)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// CUSTOM CODE STARTS HERE
+    // check if the input string is longer than 32 bytes
+    if len(inputStr) > 32 {
+      return nil, 0, errors.New("input string is longer than 32 bytes")
+    }
+
+	// setGreeting is the execution function "SetGreeting(name string)"
+    // and sets the storageKey in the string returned by hello world
+  
```

Comment:
Can we fix the indenting here?

Reply:
Unfortunately if I fix the indenting there, it messes up the actual indenting in the codeblock in the actual markdown file

Reply:
Actually I did some work around to fix it thanks